pipeline/05-finalize.R

```diff
@@ -203,262 +197,39 @@ tictoc::tic.clearlog()
-  message("Uploading final assessment results")
-
-  # Upload PIN and card-level values for full runs. These outputs are very
-  # large, so to help reduce file size and improve query performance we
-  # partition them by year, run ID, and township
-  if (run_type == "full") {
-    read_parquet(paths$output$assessment_card$local) %>%
-      mutate(run_id = run_id, year = params$assessment$year) %>%
-      group_by(year, run_id, township_code) %>%
-      arrow::write_dataset(
-        path = paths$output$assessment_card$s3,
-        format = "parquet",
-        hive_style = TRUE,
-        compression = "snappy"
-      )
-    read_parquet(paths$output$assessment_pin$local) %>%
-      mutate(run_id = run_id, year = params$assessment$year) %>%
-      group_by(year, run_id, township_code) %>%
-      arrow::write_dataset(
-        path = paths$output$assessment_pin$s3,
-        format = "parquet",
-        hive_style = TRUE,
-        compression = "snappy"
-      )
-  }
-
-
-  # 4.3. Evaluate --------------------------------------------------------------
-
-  # Upload test set performance
-  message("Uploading test set evaluation")
-  read_parquet(paths$output$performance_test$local) %>%
-    mutate(run_id = run_id) %>%
-    relocate(run_id) %>%
-    write_parquet(paths$output$performance_test$s3)
-  read_parquet(paths$output$performance_quantile_test$local) %>%
-    mutate(run_id = run_id) %>%
-    relocate(run_id) %>%
-    write_parquet(paths$output$performance_quantile_test$s3)
-
-  # Upload assessment set performance if a full run
-  if (run_type == "full") {
-    message("Uploading assessment set evaluation")
-    read_parquet(paths$output$performance_assessment$local) %>%
-      mutate(run_id = run_id) %>%
-      relocate(run_id) %>%
-      write_parquet(paths$output$performance_assessment$s3)
-    read_parquet(paths$output$performance_quantile_assessment$local) %>%
-      mutate(run_id = run_id) %>%
-      relocate(run_id) %>%
-      write_parquet(paths$output$performance_quantile_assessment$s3)
-  }
-
-
-  # 4.4. Interpret -------------------------------------------------------------
-
-  # Upload SHAP values if a full run. SHAP values are one row per card and one
-  # column per feature, so the output is very large. Therefore, we partition
-  # the data by year, run, and township
-  if (run_type == "full" && shap_enable) {
-    message("Uploading SHAP values")
-    read_parquet(paths$output$shap$local) %>%
-      mutate(run_id = run_id, year = params$assessment$year) %>%
-      group_by(year, run_id, township_code) %>%
-      arrow::write_dataset(
-        path = paths$output$shap$s3,
-        format = "parquet",
-        hive_style = TRUE,
-        compression = "snappy"
+# Wrap this block in an error handler so that the pipeline continues execution
+# even if report generation fails. This is important because the report file is
+# defined separately, so this script can't be sure that it is error-free, and
+#
```

Comment:
```suggestion
# defined separately, so this script can't be sure that it is error-free
```

Reply:
Fixed in 31dc99dc849082f004645627271dcbc8e3590003.